api/v1/submissions/PKPSubmissionHandler.inc.php

```diff
@@ -901,4 +941,281 @@ public function deletePublication($slimRequest, $response, $args)
+            return $response->withStatus(403)->withJsonError('api.submissions.403.userCantEdit');
+        }
+
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_AUTHOR, $slimRequest->getParsedBody());
+        $params['id'] = $author->getId();
+
+        $submissionContext = $request->getContext();
+        if (!$submissionContext || $submissionContext->getId() !== $submission->getData('contextId')) {
+            $submissionContext = Services::get('context')->get($submission->getData('contextId'));
+        }
+        $primaryLocale = $publication->getData('locale');
+        $allowedLocales = $submissionContext->getData('supportedSubmissionLocales');
+
+        if ($publication->getId() !== $author->getData('publicationId')) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $errors = Repo::author()->validate($author, $params, $allowedLocales, $primaryLocale);
+
+        if (!empty($errors)) {
+            return $response->withStatus(400)->withJson($errors);
+        }
+
+        Repo::author()->edit($author, $params);
+        $author = Repo::author()->get($author->getId());
+
+        return $response->withJson(
+            Repo::author()->getSchemaMap()->map($author),
+            200
+        );
+    }
+
+    /**
+     * Delete one of this publication's contributors
+     *
+     * @param $slimRequest Request Slim request object
+     * @param $response Response object
+     * @param array $args arguments
+     *
+     * @return Response
+     */
+    public function saveContributorsOrder($slimRequest, $response, $args)
+    {
+        $params = $slimRequest->getParsedBody();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+
+        $publication = Repo::publication()->get((int) $args['publicationId']);
+
+        if (!$publication) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return $response->withStatus(403)->withJsonError('api.publications.403.submissionsDidNotMatch');
+        }
+
+        if (!empty($params['sortedAuthors'])) {
+            foreach ($params['sortedAuthors'] as $author) {
+                $newAuthor = Repo::author()->get((int) $author['id']);
+
+                if (!$newAuthor) {
+                    return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+                }
+
+                if ($publication->getId() !== $newAuthor->getData('publicationId')) {
+                    return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+                }
+
+                $newSeq['seq'] = $author['seq'];
+
+                Repo::author()->edit($newAuthor, $newSeq);
+            }
+        }
+
+        return $response->withJson($publication->getId());
```

Comment:
I'm ok with keeping this API endpoint, but another option is for the `ContributorsListPanel` to reuse the `editContributor` API endpoint. In this scenario, you'd loop over the list panel items and send an edit request for each author with a change to the `seq`. It would be slower for large author lists, but also one less API endpoint. Up to you.

Reply:
If you keep it, it should be `PUT` not `POST`.

Reply:
Changed it yo PUT